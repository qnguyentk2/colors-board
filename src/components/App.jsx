import React from 'react';
import { Router, Switch } from 'react-router-dom';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createBrowserHistory } from 'history';
// import ErrorBoundary from '@components/ErrorBoundary/ErrorBoundary';
import allReducers from '@reducers';
import PageLayout from '@components/PageLayout/PageLayout';
import { GlobalProvider, GlobalConsumer } from '@contexts/GlobalContext';

const CACHE = {
  STATE: {}
};
const state = JSON.parse(JSON.stringify(CACHE.STATE));
const history = createBrowserHistory();
const store = createStore(allReducers, state, applyMiddleware(thunk));

const App = () => (
  <Provider store={store}>
    <GlobalProvider>
      <GlobalConsumer>
        {props => (
          <Router history={history}>
            {/* <ErrorBoundary> */}
            <Switch>
              <PageLayout {...props} />
            </Switch>
            {/* </ErrorBoundary> */}
          </Router>
          )}
      </GlobalConsumer>
    </GlobalProvider>
  </Provider>
);

export default App;
