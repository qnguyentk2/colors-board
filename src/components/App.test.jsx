/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import App from './App';

const setup = () => {
  const wrapper = shallow(<App />);

  return { wrapper };
}

describe('Test App', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

