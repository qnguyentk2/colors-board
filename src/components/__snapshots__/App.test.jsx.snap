// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <GlobalProvider>
    <ContextConsumer>
      <Component />
    </ContextConsumer>
  </GlobalProvider>
</Provider>
`;
